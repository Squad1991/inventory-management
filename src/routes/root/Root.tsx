import { css } from '@compiled/react';
import React from 'react';
import { Outlet, ScrollRestoration } from 'react-router-dom';
import PageLayout from '~src/components/common/page-layout/PageLayout';
import Footer from '~src/components/common/page-layout/footer/Footer';
import Header from '~src/components/common/page-layout/header/Header';
import Main from '~src/components/common/page-layout/main/Main';
import Navbar from '~src/components/common/page-layout/navbar/Navbar';
import Stack from '~src/components/common/primitive/Stack';

/**
 * This is a root component that wraps the entire application.
 * It contains the header, navbar, main content, and footer and slot for routes.
 */
const Root: React.FC = () => {
  return (
    <PageLayout>
      <Stack>
        <Header />
        <Navbar />
      </Stack>
      <Main css={css({ paddingInline: '12rem', paddingBlock: '1rem' })}>
        <ScrollRestoration />
        <Outlet />
      </Main>
      <Footer />
    </PageLayout>
  );
};

export default Root;
