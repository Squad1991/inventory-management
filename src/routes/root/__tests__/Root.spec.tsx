import { render } from '@testing-library/react';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import Root from '~src/routes/root/Root';

describe('Root component', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <RouterProvider
        router={createBrowserRouter([
          {
            path: '/',
            index: true,
            Component: Root,
          },
        ])}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
