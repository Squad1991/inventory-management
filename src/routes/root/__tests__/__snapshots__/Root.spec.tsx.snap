// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Root component > should match the snapshot 1`] = `
<div>
  <style
    data-cmpld="true"
  >
    ._1e0c11p5{display:grid}._bfhk1h82{background-color:var(--color-background)}
  </style>
  <div
    class="_1e0c11p5 _bfhk1h82"
  >
    <style
      data-cmpld="true"
    >
      ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
    </style>
    <div
      class="_1e0c1txw _2lx21bp4"
    >
      <style
        data-cmpld="true"
      >
        ._1rjc1k8s{padding-block:1.5rem}._18zr31sa{padding-inline:12rem}._4cvr1h6o{align-items:center}._4t3i1i8p{height:var(--header-height)}._1pbyryd4{z-index:var(--z-index-header)}._bfhku67f{background-color:#fff}._1bah1yb4{justify-content:space-between}
      </style>
      <header
        class="_1e0c1txw _1rjc1k8s _18zr31sa _4cvr1h6o _4t3i1i8p _1pbyryd4 _bfhku67f _1bah1yb4"
      >
        <style
          data-cmpld="true"
        >
          ._1wyb1k8s{font-size:1.5rem}
        </style>
        <style
          data-cmpld="true"
        >
          ._zulp1kwk{gap:.5rem}
        </style>
        <div
          class="_1e0c1txw _4cvr1h6o _zulp1kwk _1wyb1k8s"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 32C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48H48zm0 32h106c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H38c-3.3 0-6-2.7-6-6V80c0-8.8 7.2-16 16-16zm426 96H38c-3.3 0-6-2.7-6-6v-36c0-3.3 2.7-6 6-6h138l30.2-45.3c1.1-1.7 3-2.7 5-2.7H464c8.8 0 16 7.2 16 16v74c0 3.3-2.7 6-6 6zM256 424c-66.2 0-120-53.8-120-120s53.8-120 120-120 120 53.8 120 120-53.8 120-120 120zm0-208c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm-48 104c-8.8 0-16-7.2-16-16 0-35.3 28.7-64 64-64 8.8 0 16 7.2 16 16s-7.2 16-16 16c-17.6 0-32 14.4-32 32 0 8.8-7.2 16-16 16z"
            />
          </svg>
          <a
            href="/"
          >
            Movie Database
          </a>
        </div>
        <style
          data-cmpld="true"
        >
          ._1bsb13ks{width:28rem}
        </style>
        <style
          data-cmpld="true"
        >
          ._2lx2vrvc{flex-direction:row}
        </style>
        <div
          class="_1e0c1txw _2lx2vrvc _1bsb13ks"
        >
          <style
            data-cmpld="true"
          >
            ._1bsb1osq{width:100%}
          </style>
          <div
            class="_1bsb1osq css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              class=" css-g4qz9s-control"
            >
              <div
                class=" css-xa41c2-ValueContainer"
              >
                <div
                  class=" css-1jqq78o-placeholder"
                  id="react-select-2-placeholder"
                >
                  Search movie by name...
                </div>
                <div
                  class=" css-188ejpa-Input"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-16n3y0y-indicatorContainer"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M48 32C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48H48zm0 32h106c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H38c-3.3 0-6-2.7-6-6V80c0-8.8 7.2-16 16-16zm426 96H38c-3.3 0-6-2.7-6-6v-36c0-3.3 2.7-6 6-6h138l30.2-45.3c1.1-1.7 3-2.7 5-2.7H464c8.8 0 16 7.2 16 16v74c0 3.3-2.7 6-6 6zM256 424c-66.2 0-120-53.8-120-120s53.8-120 120-120 120 53.8 120 120-53.8 120-120 120zm0-208c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm-48 104c-8.8 0-16-7.2-16-16 0-35.3 28.7-64 64-64 8.8 0 16 7.2 16 16s-7.2 16-16 16c-17.6 0-32 14.4-32 32 0 8.8-7.2 16-16 16z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="_1e0c1txw _4cvr1h6o _zulp1kwk"
        >
          <svg
            data-testid="bell-icon"
            fill="currentColor"
            height="16"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
            />
          </svg>
          <svg
            data-testid="user-icon"
            fill="currentColor"
            height="16"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
        </div>
      </header>
      <style
        data-cmpld="true"
      >
        ._bfhkglyw{background-color:none}._4t3i1osq{height:100%}._17921j6v{padding-in-sidebar-list-itemne:1rem}._18m91wug{overflow-y:auto}._1rjc1rrh{padding-block:.2rem}._18zrhhrg{padding-inline:11rem}._n7zl10hv{border-bottom:1px solid var(--color-border)}
      </style>
      <nav
        class="_bfhkglyw _4t3i1osq _17921j6v _18m91wug _1rjc1rrh _18zrhhrg _n7zl10hv"
      >
        <style
          data-cmpld="true"
        >
          ._4ow8glyw{-sidebar-list-itemst-style:none}
        </style>
        <ul
          class="_4ow8glyw _1e0c1txw _2lx2vrvc"
        >
          <style
            data-cmpld="true"
          >
            ._18zr1j6v{padding-inline:1rem}._80omtlke{cursor:pointer}
          </style>
          <li
            class="_1e0c1txw _4cvr1h6o _zulp1kwk _18zr1j6v _80omtlke"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              />
            </svg>
            <a
              class=""
              href="/movies"
            >
              Home
            </a>
          </li>
          <li
            class="_1e0c1txw _4cvr1h6o _zulp1kwk _18zr1j6v _80omtlke"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M384 160c-17.7 0-32-14.3-32-32s14.3-32 32-32H544c17.7 0 32 14.3 32 32V288c0 17.7-14.3 32-32 32s-32-14.3-32-32V205.3L342.6 374.6c-12.5 12.5-32.8 12.5-45.3 0L192 269.3 54.6 406.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160c12.5-12.5 32.8-12.5 45.3 0L320 306.7 466.7 160H384z"
              />
            </svg>
            <a
              class=""
              href="/movies/trending"
            >
              Trending
            </a>
          </li>
          <li
            class="_1e0c1txw _4cvr1h6o _zulp1kwk _18zr1j6v _80omtlke"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M466.27 286.69C475.04 271.84 480 256 480 236.85c0-44.015-37.218-85.58-85.82-85.58H357.7c4.92-12.81 8.85-28.13 8.85-46.54C366.55 31.936 328.86 0 271.28 0c-61.607 0-58.093 94.933-71.76 108.6-22.747 22.747-49.615 66.447-68.76 83.4H32c-17.673 0-32 14.327-32 32v240c0 17.673 14.327 32 32 32h64c14.893 0 27.408-10.174 30.978-23.95 44.509 1.001 75.06 39.94 177.802 39.94 7.22 0 15.22.01 22.22.01 77.117 0 111.986-39.423 112.94-95.33 13.319-18.425 20.299-43.122 17.34-66.99 9.854-18.452 13.664-40.343 8.99-62.99zm-61.75 53.83c12.56 21.13 1.26 49.41-13.94 57.57 7.7 48.78-17.608 65.9-53.12 65.9h-37.82c-71.639 0-118.029-37.82-171.64-37.82V240h10.92c28.36 0 67.98-70.89 94.54-97.46 28.36-28.36 18.91-75.63 37.82-94.54 47.27 0 47.27 32.98 47.27 56.73 0 39.17-28.36 56.72-28.36 94.54h103.99c21.11 0 37.73 18.91 37.82 37.82.09 18.9-12.82 37.81-22.27 37.81 13.489 14.555 16.371 45.236-5.21 65.62zM88 432c0 13.255-10.745 24-24 24s-24-10.745-24-24 10.745-24 24-24 24 10.745 24 24z"
              />
            </svg>
            <a
              class=""
              href="/movies/recommended"
            >
              Recommended
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <style
      data-cmpld="true"
    >
      ._1rjc1j6v{padding-block:1rem}
    </style>
    <main
      class="_18zr31sa _1rjc1j6v"
    />
    <style
      data-cmpld="true"
    >
      ._y3gn1h6o{text-align:center}._zulp1j6v{gap:1rem}._1q511j6v{padding-block-start:1rem}._1bah1h6o{justify-content:center}._x3dol2jy{border-top:1px solid #e0e0e0}._85i51j6v{padding-block-end:1rem}
    </style>
    <footer
      class="_y3gn1h6o _1e0c1txw _zulp1j6v _1wyb1k8s _1q511j6v _1bah1h6o _x3dol2jy _85i51j6v"
    >
      <div
        class="_1e0c1txw _2lx21bp4 _zulp1j6v _4cvr1h6o"
      >
        <div
          class="_1e0c1txw _4cvr1h6o _zulp1kwk _1wyb1k8s"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M48 32C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48H48zm0 32h106c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H38c-3.3 0-6-2.7-6-6V80c0-8.8 7.2-16 16-16zm426 96H38c-3.3 0-6-2.7-6-6v-36c0-3.3 2.7-6 6-6h138l30.2-45.3c1.1-1.7 3-2.7 5-2.7H464c8.8 0 16 7.2 16 16v74c0 3.3-2.7 6-6 6zM256 424c-66.2 0-120-53.8-120-120s53.8-120 120-120 120 53.8 120 120-53.8 120-120 120zm0-208c-48.5 0-88 39.5-88 88s39.5 88 88 88 88-39.5 88-88-39.5-88-88-88zm-48 104c-8.8 0-16-7.2-16-16 0-35.3 28.7-64 64-64 8.8 0 16 7.2 16 16s-7.2 16-16 16c-17.6 0-32 14.4-32 32 0 8.8-7.2 16-16 16z"
            />
          </svg>
          <a
            href="/"
          >
            Movie Database
          </a>
        </div>
        <div
          class="_1e0c1txw _4cvr1h6o _zulp1kwk"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
            />
          </svg>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
